src/fedora_easyfix/utilities/producer.py

```diff
@@ -104,20 +107,34 @@ def write_index_to_local_json(self):
             with open("tickdata.json", "w") as tickfile:
                 tickfile.write(tickdata)
             statdcrt.section("Indexing complete!")
-        except Exception as expt:
+        except PermissionError as expt:
             statdcrt.failure("Could not index tickets")
             statdcrt.general("Please check if appropriate permissions are available to write in the directory")
             exit()
 
 
 def mainfunc():
+    statdcrt.section("Indexing tickets...")
     try:
         prodobjc = Producer()
         prodobjc.check_repolist_version_and_start()
-    except Exception as expt:
+    except KeyError as expt:
         statdcrt.failure("Could not index tickets")
         statdcrt.general("Please check if the environment variables are configured properly")
         exit()
+    except NewConnectionError as expt:
+        statdcrt.failure("Could not index tickets")
+        statdcrt.general("Please check if the repository listing is available in the specified location")
+        exit()
```

Comment:
Similar issue as `time.time`, etc. There's `sys.exit()` and the builtin `exit()` which differ slightly  – AIUI the latter is intended for interactive Python shells, it doesn't let you specify exit codes. Again, I'd `import sys` and use `sys.exit()` later to make this clear.